pkg/cloudprovider/cloudprovider.go

```diff
@@ -276,23 +252,6 @@ func (c *CloudProvider) resolveNodeClassFromNodePool(ctx context.Context, nodePo
 	return nodeClass, nil
 }
 
-// TODO @joinnis: Remove this handling for NodeTemplate resolution when we remove v1alpha5
-func (c *CloudProvider) resolveNodeTemplate(ctx context.Context, raw []byte, objRef *v1alpha5.MachineTemplateRef) (*v1alpha1.AWSNodeTemplate, error) {
-	nodeTemplate := &v1alpha1.AWSNodeTemplate{}
-	if objRef != nil {
-		if err := c.kubeClient.Get(ctx, types.NamespacedName{Name: objRef.Name}, nodeTemplate); err != nil {
-			return nil, fmt.Errorf("getting providerRef, %w", err)
-		}
-		return nodeTemplate, nil
-	}
-	aws, err := v1alpha1.DeserializeProvider(raw)
-	if err != nil {
-		return nil, err
-	}
-	nodeTemplate.Spec.AWS = lo.FromPtr(aws)
-	return nodeTemplate, nil
-}
-
 func (c *CloudProvider) resolveInstanceTypes(ctx context.Context, nodeClaim *corev1beta1.NodeClaim, nodeClass *v1beta1.EC2NodeClass) ([]*cloudprovider.InstanceType, error) {
```

Comment:
IIRC, I think that there's some handling of the provisioner name label key in instanceToNodeClaim that we can drop now that we are dropping alpha